thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1433,33 +1443,22 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
                     Owned<IKeyIndex> keyIndex = activity.createPartKeyIndex(partNo, copy, false);
                     partKeySet->addIndex(keyIndex.getClear());
                 }
-                keyManager = createKeyMerger(helper->queryIndexRecordSize()->queryRecordAccessor(true), partKeySet, 0, nullptr, helper->hasNewSegmentMonitors(), false);
+                keyManager = createKeyMerger(helper->queryIndexRecordSize()->queryRecordAccessor(true), partKeySet, 0, &contextLogger, helper->hasNewSegmentMonitors(), false);
                 setupTranslation(0, 0, *keyManager);
             }
             processRows(processing, 0, keyManager);
         }
-        virtual void getFileStats(std::vector<OwnedPtr<CRuntimeStatisticCollection>> & fileStats, unsigned startOffset) override
+        virtual void getFileStats(std::vector<OwnedPtr<CRuntimeStatisticCollection>> & fileStats, unsigned startOffset) const override
         {
-            if (keyManager)
-            {
-                for (size_t i=0; i<parts.size(); i++)
-                {
-                    if (isSuper)
-                    {
-                        unsigned subfile = subFileNum[i];
-                        (*keyManager).mergeStats(*fileStats[startOffset+subfile]);
-                    }
-                    else
-                        (*keyManager).mergeStats(*fileStats[startOffset]);
-                }
-            }
+            // CKeyLookupMergeHandler is not tracking the stats at the part level and that means it is not possible to associate the stats with a  logical file level.
+            // Superfile: all stats will be associated with the first sub index file.  For non-super, it makes no difference (there is only one index file).
+            // (Future: make necessary changes to createKeyMerger to track stats at the part level/logical level.)
```

Comment:
can you open a JIRA and link it to this one?

Reply:
Will do. TODO: create ticket once jira is working again.

Reply:
jira: https://track.hpccsystems.com/browse/HPCC-30599